source/include-link.txt

```diff
@@ -0,0 +1,209 @@
+   $ cmake --build build
+   [ 50%] Building CXX object CMakeFiles/view_and_value.dir/<path-to-mongo-cxx-driver-sources>/examples/bsoncxx/view_and_value.cpp.o
+   [100%] Linking CXX executable view_and_value
+   [100%] Built target view_and_value
+   $ ./build/view_and_value
+   { "team" : "platforms", "id" : { "$oid" : "66f4be6fef9eb8b9240619f0" }, "members" : [ "tyler", "jason", "drew", "sam", "ernie", "john", "mark", "crystal" ] }
+   Got key, key = team
+   Got String!
+   Got key, key = id
+   Got ObjectId!
+   Got key, key = members
+   Got Array!
+   array element: tyler
+   array element: jason
+   array element: drew
+   array element: sam
+   array element: ernie
+   array element: john
+   array element: mark
+   array element: crystal
+   as expected, we have a team
+   document has 3 keys.
+   document keys are:
+   team
+   id
+   members
+
+If you encounter errors, you may need to specify additional options to the initial CMake command. The specific options depend on your particular environment, however there are two options that address most common issues. For instance, if your compiler does not default to at least C++17, use the ``-DCMAKE_CXX_STANDARD=17`` CMake option. If you installed the driver to a non-standard location, you may need to specify the ``-DCMAKE_PREFIX_PATH=/<path-to-mongo-cxx-driver-installation>`` option. For example:
+
+.. code-block:: none
+
+   $ cmake -Bbuild -DCMAKE_CXX_STANDARD=17 -DCMAKE_PREFIX_PATH=/opt/mongodb/cxx-driver
+
+Without Driver Installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Alternatively, you can use CMake's ``add_subdirectory`` command without installing the {+driver-short+}. This is an advanced technique that, unlike the  ``find_package`` command, does not support specifying version constraints. There are additional drawbacks associated with ``add_subdirectory``, but discussion of those is beyond the scope of this document.
+
+.. note::
+
+   You may need to adjust build flags for your project to
+   accommodate this approach. Your project's invocation of CMake must include any flags or
+   options that you would have passed to the {+driver-short+} as part of its build.
+
+To use the ``add_subdirectory`` command, create a ``CMakeLists.txt`` file in your project directory. The following example creates a ``CMakeLists.txt`` file in the ``/home/user/project2`` project directory that uses ``add_subdirectory``:
+
+.. literalinclude:: /includes/cmake_without_driver_installation.txt
+  :caption: /home/user/project2/CMakeLists.txt
+  :start-after: -- sphinx-include-start --
+
+.. note::
+
+   Note that in this approach the CMake target is referred to simply
+   as ``bsoncxx_shared``, without the ``mongo::`` namespace. The namespace
+   is added as part of the CMake module installation, which is not performed
+   in this approach.
+
+Then, you can build and run your project. The following shows a truncated example of the CMake output and the program execution output immediately following:
```

Comment:
S: same suggestions here about breaking up the code example with multiple commands/outputs